pkg/virt-api/rest/subresource.go

```diff
@@ -307,20 +311,12 @@ func (app *SubresourceAPIApp) StartVMRequestHandler(request *restful.Request, re
 		patchType = types.JSONPatchType
 
 		needsRestart := false
-		vmi, err := app.VirtCli.VirtualMachineInstance(namespace).Get(name, &k8smetav1.GetOptions{})
-		if err != nil {
-			if !errors.IsNotFound(err) {
-				response.WriteError(http.StatusInternalServerError, err)
-				return
-			}
-		} else {
-			if (runStrategy == v1.RunStrategyRerunOnFailure && vmi.Status.Phase == v1.Succeeded) ||
-				(runStrategy == v1.RunStrategyManual && vmi.IsFinal()) {
-				needsRestart = true
-			} else if runStrategy == v1.RunStrategyRerunOnFailure && vmi.Status.Phase == v1.Failed {
-				response.WriteError(http.StatusBadRequest, fmt.Errorf("runstrategy rerunonerror does not support starting VM from failed state"))
-				return
-			}
+		if (runStrategy == v1.RunStrategyRerunOnFailure && vmi.Status.Phase == v1.Succeeded) ||
+			(runStrategy == v1.RunStrategyManual && vmi.IsFinal()) {
+			needsRestart = true
+		} else if runStrategy == v1.RunStrategyRerunOnFailure && vmi.Status.Phase == v1.Failed {
+			response.WriteError(http.StatusBadRequest, fmt.Errorf("runstrategy rerunonerror does not support starting VM from failed state"))
+			return
 		}
 
 		if needsRestart {
```

Comment:
@stu-gott to add to the discussion of restart behavior from above, this code of checking wouldn't make sense any more, right? I mean, there shouldn't exist a VMI at this point, so we could remove this?

Reply:
The check being done above specifically makes an exception for VMI's in the succeeded and failed states. so as-written, I think this block still applies.